website/docs/cloud-docs/integrations/kubernetes/ops-v2-migration.mdx

```diff
@@ -43,33 +43,220 @@ Next, backup the workspace resources.
+| `spec.outputs.[*].key` | `spec.outputs.[*].name` | Output names should be included in the module. This is a reference to the output variables produced by the module. |
+| `status.workspaceID` OR `metadata.namespace-metadata.name` | `spec.workspace.id` OR `spec.workspace.name` | The workspace where the module will be executed. The workspace must be within the same organization. |
+
+Below is an example migration of a `Module` between v1 and v2 of the operator:
+
+<CodeBlockConfig filename="v1.yaml">
+
+```yaml
+apiVersion: app.terraform.io/v1alpha1
+kind: Workspace
+metadata:
+  name: migration
+spec:
+  module:
+    source: app.terraform.io/org-name/module-name/provider
+    version: 0.0.42
+  variables:
+    - key: username
+      value: "user"
+      hcl: true
+      sensitive: false
+      environmentVariable: false
+    - key: SECRET_KEY
+      value: "s3cr3t"
+      hcl: false
+      sensitive: false
+      environmentVariable: true
+```
+
+</CodeBlockConfig>
+
+In v2 of the operator, the workspace and module are manage by separate controllers.
+
+<CodeBlockConfig filename="workspace-v2.yaml">
+
+  ```yaml
+  apiVersion: app.terraform.io/v1alpha2
+  kind: Workspace
+  metadata:
+    name: migration
+    spec:
+      terraformVariables:
+        - name: username
+          value: "user"
+          hcl: true
+          sensitive: false
+      environmentVariables:
+        - name: SECRET_KEY
+          value: "s3cr3t"
+          hcl: false
+          sensitive: false
+  ```
+
+</CodeBlockConfig>
+
+<CodeBlockConfig filename="module-v2.yaml">
+
+```yaml
+apiVersion: app.terraform.io/v1alpha2
+kind: Module
+metadata:
+  name: migration
+spec:
+  name: operator
+  module:
+    source: app.terraform.io/org-name/module-name/provider
+    version: 0.0.42
+  workspace:
+    name: migration
+```
+
+</CodeBlockConfig>
+
+## Upgrade the operator
+
+Download Workspace CRD patch A:
 
 ```shell-session
-$ cd terraform-cloud-operator/docs/migration/crds
+$ curl -sO https://raw.githubusercontent.com/hashicorp/terraform-cloud-operator/main/docs/migration/crds/workspaces_patch_a.yaml
 ```
 
 View the changes that **patch A** will apply to the workspace CRD.
 
 ```shell-session
-$ kubectl diff -f workspaces_patch_a.yaml
+$ kubectl diff --filename workspaces_patch_a.yaml
 ```
 
 Patch the workspace CRD with **patch A**. This patch adds `app.terraform.io/v1alpha2` support, but excludes `.status.runStatus` because it has a different format in `app.terraform.io/v1alpha1` and causes JSON unmarshalling issues.
 
+~> **Warning** Once the patch is applied, Kubernetes will convert existing `app.terraform.io/v1alpha1` custom resources to `app.terraform.io/v1alpha2` according to the updated schema and v1 of the operator will no longer serve custom resources. You must update existing custom resources to satisfy the current schema. Refer to the tables above for a mapping of resource fields between v1 and v2 of the operator.
```

Comment:
I think we can shorten this to:
```suggestion
!> **Upgrade warning**: Once you apply a patch, Kubernetes converts existing `app.terraform.io/v1alpha1` custom resources to `app.terraform.io/v1alpha2` according to the updated schema, meaning that v1 of the operator can no longer serve custom resources. Before patching, update your existing custom resources to satisfy the v2 schema requirements. [Learn more](#manifest-schema-migration).
```

I also suggested using the [suggested callout styling for upgrade warnings](https://github.com/hashicorp/engineering-docs/blob/main/writing/style-guide.md#labels)